tests/test_mql.py

```diff
@@ -1001,3 +1105,183 @@ def test_parse_mql_arbitrary_functions(
 ) -> None:
     result = parse_mql(mql_string)
     assert result == metrics_query
+
+
+curried_arbitrary_function_tests = [
+    pytest.param(
+        "topK(10)(transaction.duration)",
```

Comment:
It seems like, based on the grammar, this test should fail, since this doesn't match a `curried_aggregate`. I think it does pass because you have some smart logic in the Visitor class, but it would be cleaner if it was the grammar that was making this work.

Reply:
It matches `curried_arbitrary_function`